### ISSUE TITLE
feat: better output control

### ISSUE THREAD

**Issue by osterman** [mumoshu/variant#40]
## what
* when running a command the output is first emitted as loglevel `INFO` as well as directly to `stdout`

## why 
* I think `stdout` should just be standard output. We are after all shelling out to other tools (e.g. `terraform`) that we'd like to see the output.
* capturing the `stdout` for *optionally* logging to a **file** is cool. I like that idea, but not sure how to best reconcile it. with the duplicate output.

## example

```sh
#!/usr/bin/env variant

tasks:
  ls:
    description: "test"
    script: |
      /usr/bin/ls -l 
```

Produces this output.
```
NFO[0000] total 24                                      stream=stdout
INFO[0000] -rw-r--r-- 1 CORP\erik.osterman CORP\domain users  95 Jan 14 04:15 DESCRIPTION  stream=stdout
INFO[0000] -rw-r--r-- 1 CORP\erik.osterman CORP\domain users  11 Jan 14 04:15 LICENSE  stream=stdout
INFO[0000] -rw-r--r-- 1 CORP\erik.osterman CORP\domain users 382 Dec 21 16:10 Makefile  stream=stdout
INFO[0000] -rw-r--r-- 1 CORP\erik.osterman CORP\domain users   2 Jan 14 04:17 RELEASE  stream=stdout
INFO[0000] -rwxr-xr-x 1 CORP\erik.osterman CORP\domain users 324 Jan 14 04:49 testme  stream=stdout
INFO[0000] -rw-r--r-- 1 CORP\erik.osterman CORP\domain users   6 Jan 14 13:02 VERSION  stream=stdout
total 24
-rw-r--r-- 1 CORP\erik.osterman CORP\domain users  95 Jan 14 04:15 DESCRIPTION
-rw-r--r-- 1 CORP\erik.osterman CORP\domain users  11 Jan 14 04:15 LICENSE
-rw-r--r-- 1 CORP\erik.osterman CORP\domain users 382 Dec 21 16:10 Makefile
-rw-r--r-- 1 CORP\erik.osterman CORP\domain users   2 Jan 14 04:17 RELEASE
-rwxr-xr-x 1 CORP\erik.osterman CORP\domain users 324 Jan 14 04:49 testme
-rw-r--r-- 1 CORP\erik.osterman CORP\domain users   6 Jan 14 13:02 VERSION
```

**Comment by mumoshu** [comment]
> I think stdout should just be standard output. We are after all shelling out to other tools (e.g. terraform) that we'd like to see the output.

Yep, it makes sense. I thought I've changed the default behavior to `log-to-stderr` though, so that only `/usr/bin/ls -l` output is streamed to the stdout, while the INFO logs are sent to STDERR.

https://github.com/mumoshu/variant/commit/c2f759934c6fb655ab8d3d0d2601bd344c172632

Which version of variant did you try, the latest release or the master(`go get -u` or whatever)?

**Comment by mumoshu** [comment]
> capturing the stdout for optionally logging to a file is cool. I like that idea, but not sure how to best reconcile it. with the duplicate output.

Something that works like `variant foo bar | tee log.file`, right?

**Comment by osterman** [comment]
> Something that works like `variant foo bar | tee log.file`, right?

Yep... exactly (although also capturing `stderr`)

**Comment by osterman** [comment]
I'm running https://github.com/mumoshu/variant/releases/tag/v0.10.0, which is a different commit.

**Comment by osterman** [comment]
@mumoshu I updated this issue to better reflect the decision we arrived at.

**Comment by mumoshu** [comment]
@osterman Thanks!

Would you like to set log-level per destination, like WARN for stdout/stderr(its INFO by default), INFO for the log file 1? Then, do you have any idea what are the canonical flags for the configuration?

What I came up is `./mycmd --loglevel WARN --logtofile mylog.txt=INFO`, but not sure if it looks good or not.

**Comment by osterman** [comment]
> Would you like to set log-level per destination, like WARN for stdout/stderr(its INFO by default), INFO for the log file 1?

Not a strict requirement for first implementation. Just a "nice to have"

> ./mycmd --loglevel WARN --logtofile mylog.txt=INFO

Filenames can get nasty, especially when working on Windows, etc.

Thus, my suggestion is to always make the predictable key the first one: `--logtofile INFO:mylog.txt`

I can't think of a great reference example. Only one that comes to mind is `socat` which does something similar with it's arguments.

https://github.com/craSH/socat/blob/master/EXAMPLES#L159
